### PR TITLE
[Feature][Zeta] Support slf4j mdc tracing job id output into logs

### PR DESCRIPTION
**Pull request by hailin0**
### Purpose of this pull request

[Feature][Zeta] Support slf4j mdc tracing job id output into logs

reference
https://www.slf4j.org/manual.html#mdc
https://logging.apache.org/log4j/2.x/manual/thread-context.html#configuration
https://logging.apache.org/log4j/2.x/manual/layouts.html#PatternLayout

subtask of https://github.com/apache/seatunnel/issues/2725

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?

Added

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by hailin0** [comment]
img_v3_02e2_5a13324d-dd6b-4d3c-94d5-d1f0fcb443fg
img_v3_02e2_e03e4732-5174-42ac-a91f-72cab838f89g
img_v3_02e2_a24373b2-777f-433d-99b7-130eb11c2f0g
img_v3_02e2_46670f2e-388e-40ec-9217-aa26034df05g

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/pom.xml [hunk 1]
```diff
@@ -107,6 +107,16 @@
             <artifactId>hadoop-aliyun</artifactId>
             <version>${hadoop-aliyun.version}</version>
             <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.slf4j</groupId>
+                    <artifactId>slf4j-log4j12</artifactId>
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/pom.xml:113]: img_v3_02e9_d9119e5c-9e1f-4b6d-9f87-b9d2aa80403g

Comment (hailin0) [reply seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/pom.xml:113]: img_v3_02e9_7f329aa6-f531-444e-80dd-63397ffa35ag

File: seatunnel-core/seatunnel-starter/src/main/bin/seatunnel-cluster.sh [hunk 2]
```diff
@@ -83,6 +83,7 @@ done
 
 # Log4j2 Config
 JAVA_OPTS="${JAVA_OPTS} -Dlog4j2.contextSelector=org.apache.logging.log4j.core.async.AsyncLoggerContextSelector"
+JAVA_OPTS="${JAVA_OPTS} -Dlog4j2.isThreadContextMapInheritable=true"
```
Comment (hailin0) [inline seatunnel-core/seatunnel-starter/src/main/bin/seatunnel-cluster.sh:86]: https://logging.apache.org/log4j/2.x/manual/thread-context.html#configuration

**Review by Hisoka-X — APPROVED** [review]

**Review by Carl-Zhou-CN — APPROVED** [review]
+1

**Comment by arshadmohammad** [comment]
Good feature, Thanks @hailin0 for working on this.  
related to this feature, is there anyway to get all the logs corresponding to job id, using API or command ?

**Comment by hailin0** [comment]
> Good feature, Thanks @hailin0 for working on this. related to this feature, is there anyway to get all the logs corresponding to job id, using API or command ?

@arshadmohammad 
The logs of distributed programs(job) are scattered on different machines. What are your thoughts?

**Comment by arshadmohammad** [comment]
I was looking for functionality similar to mapreduce and yarn, all mapreduce jobs have separate logs.

**Comment by hailin0** [comment]
I will do some verification later

**Pull request merged**